Implementing Dark Mode - #issue4

saumyapandey1998 opened the pull request:
This patch used both local as database storage to store user preference for dark mode. Local storage is being used for anonymous users (users that are not logged in) Database storage used for authenticated users.
CSS has been modified to improve readibility
A minimal  button has been addded to the top right corner.

This PR attempts to solve #4 

I will add screenshots to the PR.
<img width="1509" alt="Screenshot 2025-07-06 at 19 21 24" src="https://github.com/user-attachments/assets/318154ac-2b94-4c23-a00a-8707e21a09c0" />
<img width="1501" alt="Screenshot 2025-07-06 at 19 21 33" src="https://github.com/user-attachments/assets/e16d8a02-58ec-415d-b5ab-02df45f64a94" />
<img width="1505" alt="Screenshot 2025-07-06 at 19 21 41" src="https://github.com/user-attachments/assets/7463707a-dd99-4be1-8fbb-4b8df8d61f2c" />
<img width="1508" alt="Screenshot 2025-07-06 at 19 21 49" src="https://github.com/user-attachments/assets/1c9e0b98-2ede-4581-8fca-1ab9ee807708" />
<img width="1495" alt="Screenshot 2025-07-06 at 19 22 27" src="https://github.com/user-attachments/assets/79707751-0e8c-4ff6-a2fe-3af0531c8bf8" />
<img width="1511" alt="Screenshot 2025-07-06 at 19 23 06" src="https://github.com/user-attachments/assets/af193f72-6cb8-425e-887f-378e84842984" />
<img width="1512" alt="Screenshot 2025-07-06 at 19 23 14" src="https://github.com/user-attachments/assets/c72e770e-5ae0-4175-a053-62fbc4adcd23" />

Commit:
Implementing Dark Mode - Issue 4

This patch used both local as database storage to store user preference for dark
mode. Local storage is being used for anonymous users (users that are not logged in)
Database storage used for authenticated users.
CSS has been modified to improve readibility
A minimal  button has been addded to the top right corner.

I will add screenshots to the PR.

nadyagrishina commented:
Everything looks great from the screenshots! 🖤 Really clean dark mode styling — I love that you’ve covered all the screens and handled both localStorage + DB 🙌

One small note: the background feels a bit noisy/rough, and the teal color (Reflect-Diary logo) gets a little lost against it. Maybe a slightly smoother background or a more contrasty accent color would help?

Overall though — super solid work so far!

saumyapandey1998 commented:
Hi @nadyagrishina ! The grainy background is due to:
`    background-image: url("../images/groovepaper.png");
`
In the upcoming patch I have removed it and changed accent color to a brighter green to maintain the theme of the project without compromising on visibility

Commit:
Implementing Dark Mode - #issue4

Removed grainy background on the dashboard
Improved over all visibility
removed redundant code

nadyagrishina commented:
Hey again! Just wanted to say — the dark mode is working really smoothly, amazing job 🖤
I noticed that on the entry details page, some of the lighter text is a bit hard to read in dark mode.
I can totally fix it myself if you're busy — or if you'd like to polish it yourself, feel free to push a quick update.
Either way, thank you so much again for this contribution, it's awesome 💖
Thanks again, this is a fantastic contribution ✨

saumyapandey1998 committed:
Issue 4 Implementing Dark Mode

- Fixed css for  entry details page
